vault/external_tests/delegated_auth/delegated_auth_test.go

```diff
@@ -0,0 +1,396 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package delegated_auth
+
+import (
+	"context"
+	"fmt"
+	paths "path"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/builtin/credential/userpass"
+	"github.com/hashicorp/vault/helper/testhelpers"
+	"github.com/hashicorp/vault/helper/testhelpers/teststorage"
+	"github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/vault"
+	"github.com/stretchr/testify/require"
+)
+
+func buildDaError(defaults map[string]string, d *framework.FieldData) *logical.RequestDelegatedAuth {
+	fieldDataOrDefault := func(field string, d *framework.FieldData) string {
+		if val, ok := d.GetOk(field); ok {
+			return val.(string)
+		}
+
+		return defaults[field]
+	}
+
+	accessor := fieldDataOrDefault("accessor", d)
+	path := fieldDataOrDefault("path", d)
+	username := fieldDataOrDefault("username", d)
+	password := fieldDataOrDefault("password", d)
+
+	loginPath := paths.Join(path, username)
+	data := map[string]interface{}{"password": password}
+
+	return logical.NewDelegatedAuthenticationRequest(accessor, loginPath, data, nil)
```

Comment:
NOTE: Add a test case around backend error handling that is missing (the nil here).

Reply:
Added in https://github.com/hashicorp/vault/pull/23924/commits/f6d54297d7c65f049a99d7903d26e3dff55f9fdd